webapp/javascript/components/FileUploader.tsx

```diff
@@ -30,8 +30,14 @@ export default function FileUploader({ onUpload }: Props) {
           const s = JSON.parse(
             String.fromCharCode.apply(null, new Uint8Array(binaryStr))
           );
```

Comment:
I would handle this a little bit more carefully, since users may 

This is something we already discussed internally, we want in the future to validate ALL input using something like [zod](https://github.com/colinhacks/zod). But for now I think you can just do something quick and dirty and check if the required fields of a flamebearer exist.

Reply:
What would be the proposed approach? I have been looking if we could take advantage of the type definition and convert/cast the type, but I didn't find any clear way to do this, not sure if you know any. Or may be you are proposing something like:

```js
for field in ['names', 'levels', 'numTicks', 'spyName', 'format'] assert(field in flamebearer);
```

Reply:
Yeah, something like that! 

Just don't forget we need something to tell users how the file does not have the expected format. 

I know we have a notification component but I never got to use it. If you find it too out of your way, a classic `alert` can do the work TBH.

Reply:
Added some basic checks in https://github.com/pyroscope-io/pyroscope/pull/580/commits/9b3bb8089b4b2cdd83f1936209f41610a83e752a. Once https://github.com/pyroscope-io/pyroscope/pull/596 is merged these checks should be moved to flamebearer decoder I think.